src/assets/js/protocol/protocol-menu.js

```diff
@@ -0,0 +1,233 @@
+            current[i].classList.remove('selected');
+            current[i].setAttribute('aria-selected', false);
+            current[i].setAttribute('aria-expanded', false);
+        }
+
+        if (typeof _options.onMenuClose === 'function' && current.length > 0) {
+            _options.onMenuClose();
+        }
+
+        return current.length > 0;
+    };
+
+    Menu.onCloseButtonClick = function(e) {
+        e.preventDefault();
+
+        if (typeof _options.onMenuButtonClose === 'function') {
+            _options.onMenuButtonClose();
+        }
+
+        Menu.close();
+    };
+
+    Menu.toggle = function(el) {
+        var state = el.classList.contains('selected') ? true : false;
+
+        if (!state) {
+            Menu.open(el);
+        } else {
+            Menu.close();
+        }
+    };
+
+    Menu.onMouseEnter = function(e) {
+        clearTimeout(_hoverTimeout);
+
+        // Only open the panel if the user shows hover intent
+        _hoverTimeout = setTimeout(function() {
+            var current = Menu.close();
+            var animate = current ? false: true;
+
+            Menu.open(e.target, animate);
+        }, 150);
+    };
+
+    Menu.onMouseLeave = function() {
+        // Clear hover intent timer.
+        clearTimeout(_hoverTimeout);
+
+        _hoverTimeout = setTimeout(function() {
+            Menu.close();
+        }, 150);
+    };
+
+    Menu.onFocusOut = function() {
+        var self = this;
+
+        /**
+         * After an element loses focus, `document.activeElement` will always be `body` before
+         * moving to the next element. A `setTimeout` of `0` circumvents this issue as it
+         * re-queues the JavaScript to run at the end of the current excecution.
+         */
+        setTimeout(function() {
+            if (!self.contains(document.activeElement)) {
+                Menu.close();
+            }
+        }, 0);
+    };
+
+    Menu.onClickWide = function(e) {
+        e.preventDefault();
+        Menu.close();
+        Menu.open(e.target.parentNode);
+    };
+
+    Menu.onClickSmall = function(e) {
+        e.preventDefault();
+        Menu.toggle(e.target.parentNode);
+    };
+
+    Menu.isWideViewport = function() {
+        return _mqWideNav.matches;
+    };
+
+    Menu.handleState = function() {
+        _mqWideNav = matchMedia('(min-width: 768px)');
+
+        _mqWideNav.addListener(function(mq) {
+            Menu.close();
+
+            if (mq.matches) {
+                Menu.unbindEventsSmall();
+                Menu.bindEventsWide();
+            } else {
+                Menu.unbindEventsWide();
+                Menu.bindEventsSmall();
+            }
+        });
+
+        if (Menu.isWideViewport()) {
+            Menu.bindEventsWide();
+        } else {
+            Menu.bindEventsSmall();
+        }
+    };
+
+    Menu.bindEventsWide = function() {
+        var items = document.querySelectorAll('.mzp-c-menu-category.mzp-has-drop-down');
```

Comment:
I know you're sick of looking at this....

But what do you think of using .mzp-js- classes with no styles on them to handle all behaviour? Is it worth doing the extra abstraction to we can use it with dramatically different styles? 

I'm thinking of the footer, but that will probably not have significant style differences. Not sure where else we might want to use this.

Reply:
Good call, something like `.mzp-js-expandable` may be appropriate perhaps. The menu does work without JS enabled, but I think just for purposes of toggling state (as opposed to styling), this would be ok.

Reply:
It does mean we now have two classes (one for JS behavior and one for styling) which is more verbose, but if we want to keep things totally separate then fine by me.